downloads/swan-lake-release-notes/swan-lake-2201.12.0/RELEASE_NOTE.md

```diff
@@ -50,6 +50,69 @@ To view bug fixes, see the [GitHub milestone for Swan Lake Update 12 (2201.12.0)
 
 ### Deprecations
 
+### New features
+
+#### `http` package
+
+- Introduced the default HTTP listener, allowing multiple services from different Ballerina packages to be attached to the same listener. By default, the listener runs on port `9090`
+
+  ```ballerina
+  import ballerina/http;
+
+  listener http:Listener httpListener = http:getDefaultListener();
+
+  service /api/v1 on httpListener {
+
+      resource function get greeting() returns string {
+          return "Hello, World from Service 1!";
+      }
+  }
+  ```
+
+  The port and configuration of the default listener can be customized in `Config.toml` as follows:
+
+  ```toml
+  [ballerina.http]
+  defaultListenerPort = 8080
+
+  [ballerina.http.defaultListenerConfig]
+  httpVersion = "1.1"
+
+  [ballerina.http.defaultListenerConfig.secureSocket.key]
+  path = "resources/certs/ballerinaKeystore.p12"
+  password = "ballerina"
+  ```
+- Added the following static analysis rules to the Ballerina scan tool
+
+  | RuleID           | Rule Kind     | Rule Description                                       |
+  |------------------|---------------|--------------------------------------------------------|
+  | ballerina/http:1 | VULNERABILITY | Avoid allowing default resource accessor               |
+  | ballerina/http:2 | VULNERABILITY | Avoid permissive Cross-Origin Resource Sharing         |
+
+- Introduced an API to convert an `http:Response` object into a `http:StatusCodeRecord`, which includes the status code, body, and headers
+
+  ```ballerina
+  http:Client clientEP = check new ("localhost:9090/api");
+  http:Response res = check clientEP->/path;
+  http:StatusCodeRecord statusCodeRes = check res.getStatusCodeRecord();
+  ```
+
+### Improvements
+
+#### `http` package
+
+- Moved SSL context creation to the client initialization process for better performance
```

Comment:
This is an internal implementation detail, right? Should we include it here? Maybe, we can just mention performance improvements for HTTP client?

Reply:
I think it might be misleading to mention it as a generic performance improvement. This is basically depends on the implementation(whether they are using a trust store with multiple certificates and whether they are initialising the client with in the resource function or at root level)

Reply:
Then let's mention that. I don't think `Moved SSL context creation to the client initialization process` makes sense to an external developer.

Reply:
Do you mean something like this:
```suggestion
- Improved HTTP client performance by moving SSL context creation to client initialization, particularly when using a trust store with multiple certificates.
```